Gomoku.java

```diff
@@ -0,0 +1,215 @@
+       int menuChoice = userInput.nextInt();
+       GameLoop(Player,board,userInput,menuChoice);
+    }
+    
+    
+    public static void GameLoop(int Player, int[][]board, Scanner userInput, int menuChoice)
+    {
+       while (WinCondition(ColCheck(board),RowCheck(board),DiagCheck(board))==false)
+       {
+           Player = PlayerToken(Player);//updates current player
+           
+           int[] User = UserInput(userInput,board,Player,menuChoice);//take user input validates user input
+ 
+           board = Update(board,User,Player);//updates the board
+ 
+           PrintBoard(board); //print updated board
+           
+       WinCondition(ColCheck(board),RowCheck(board),DiagCheck(board));//check if a win condition is met
+       }
+       System.out.print("Player"+Player+"wins!!");
+    }
+    /*
+    * checks is 5 in a row is found either vertically, Horizontally, or Diagonally
+    * if any of these are true, then the game ends
+    */
+    public static boolean WinCondition(boolean X, boolean Y, boolean Diagon)
+    {
+       if(X==false&&Y==false&&Diagon==false)
+           return false;
+       else
+           return true;
+    }
+ 
+    /*
+    * Checks the columns for 5 sequential inputs that are equal
+    */
+    public static boolean ColCheck(int[][]board)
+    {
+       for(int i = 0;i<board.length;i++)
+       {
+           int count =0;//resets count when moving to a new column
+           int currentNumber=board[0][i];//sets count to the top of the new column
+           for(int j = 0; j<board[i].length;j++)//sequences down the column
+           {
+               /*
+               * sequences down the columns counting up if the next number in the sequence is 
+                * equal to the previous number in the sequence. Ends the game if 5 numbers in sequence
+               * are equal.
+               * 
+                * Resets to 0 if not equal.
+               */
+               if(board[j][i]==currentNumber && currentNumber !=0)
+               {
+                  count++;
+                  if(count == 5)
+                  {
+                      return true;
+                  }
+               }
+               else
+               {
+                  count=1;
+               }
+               currentNumber=board[j][i];//updates the previously checked number
+           }
+       }
+       return false;
+    }
+ 
+    public static boolean RowCheck(int[][]board)
+    {
+       
+       for(int i = 0;i<board.length;i++)
+       {
+           int count =0;
+           int currentNumber=board[i][0];
+           for(int j = 0; j<board[i].length;j++)
+           {
+               if(board[i][j]==currentNumber && currentNumber!=0)
+               {
+                  count++;
+                  if(count == 5)
+                  {
+                      return true;
+                  }
+               }
+               else
+               {
+                  count=1;
+               }

```

Comment:
original logic was a reset to 0 but it should have reset to one accounting for the new current number